### ISSUE TITLE
View archived records in AssetAdmin

### ISSUE THREAD

**Issue by chillu** [silverstripe/silverstripe-versioned#38]
# Overview

The term "archive" implies that records can be retrieved again afterwards. That's the case for pages, where a search can filter by "including archived", and then restore draft. It's not currently possible for other versioned records managed in a ModelAdmin or GridField.

# Acceptance Criteria

 * Search results include archived records which don't exist in draft of live stages
 * New mode switch applies in Page, ModelAdmin, AssetAdmin and versioned GridField views
 * Mode switches continue applying with AssetAdmin searches

# Notes

 * Needs to be accessible
 * Should be a react component that can be reused in Entwine-based sections (see https://github.com/silverstripe/silverstripe-versioned/issues/38)
 * Might be blocked by a switch from `react-bootstrap` to `reactstrap` to avoid over-investing?
 * Assumes that search performance won't be greatly affected by a left join to `<MyRecord>_Versions`
 * Might require adjustments in `Versioned->augmentSQL()`
 * Will need to include popover work with checkboxes
 * Excludes any changes to the current tree behaviour or pages section
 * There's no need to retain the state in the URL, a page reload will show the default view again
 * Should work with [versioned GraphQL](https://app.zenhub.com/workspace/o/silverstripe/silverstripe-graphql/issues/30)

# Related

 * [View archived records in ModelAdmin and GridFields](https://github.com/silverstripe/silverstripe-versioned/issues/91)
 * [Trashcan](https://github.com/silverstripe/silverstripe-framework/issues/6202)
 * [Pages search should show filter criteria](https://github.com/silverstripe/silverstripe-cms/issues/1891)
 * [Update CMS search functionality to match the newer search pattern](https://github.com/silverstripe/silverstripe-cms/issues/1777)

![image](https://user-images.githubusercontent.com/111025/29637695-12236022-88a9-11e7-905d-f514b2aaaecc.png)

**Comment by clarkepaul** [comment]
An MVP could be to not use the gridfield search (in gridfield header) but rather use the modelAdmin search where we could have a dropdown to show archived items. In the long term unfortunately I don't think this is a good pattern though as when user tested no one could find how to retrieve archived pages.

**Comment by clarkepaul** [comment]
This is a quick rough idea (icon/text could be different). Archived items could be viewed in context of the related items of the same type. This option would still allow us to add a global trash area at some stage where you could see everything deleted/archived in one place.
![view archived](https://user-images.githubusercontent.com/555033/30840600-bbd4bb08-a2d4-11e7-8ce4-52cfb27bee7e.png)

**Comment by clarkepaul** [comment]
Other icons which might be appropriate 
![image](https://user-images.githubusercontent.com/555033/30886352-32567b9c-a373-11e7-9f8f-76f2f7e243f9.png)

**Comment by newleeland** [comment]
**Potential design**
![poc_pl_gridfield_search filters](https://user-images.githubusercontent.com/8085295/32422927-3329d618-c308-11e7-804a-c1498b975c6f.png)

**Comment by flamerohr** [comment]
a worthy note, we can't get batch actions for archived pages :)
since the only way to see archived pages is through search and search is enforced to be list view and list view does not have batch actions

**Comment by clarkepaul** [comment]
Please use the Bootstrap dropdown as that accepts forms within them and it "should" be accessible (not tested properly).

**Comment by chillu** [comment]
Marking this as `impact/high` so it stays on our radar, it's a commitment we have for co-fund.

**Comment by chillu** [comment]
Estimated as AssetAdmin work only, Ingo to split out rest of work

Potential parts:

 * Popover React component with checkboxes (accessible), integrated into GridField (in ModelAdmin)
 * Versioned GraphQL for assets
 * Implement this in AssetAdmin
 * Versioned GridField ORM searches (new component?) for non-react GridField (pages, modeladmin, ...)

Note: Excludes pages (already works, just less powerful and different UI, search rather than popover based)

**Comment by chillu** [comment]
After further discussion, we've split out the work by feature area rather than technical topic: See [View archived records in ModelAdmin and GridFields](https://github.com/silverstripe/silverstripe-versioned/issues/38))

@flamerohr Can you please assess the impact and complexity of moving from `react-bootstrap` to `reactstrap` in a Github issue? That might block the implementation of this card, since we should reuse a popopver component rather than writing our own one.

@newleeland We still need designs for where this popover trigger sits in the AssetAdmin UI, your designs above are only for ModelAdmin and GridField. Can you please prioritise this?

**Comment by flamerohr** [comment]
I've done some investigation for switching to `reactstrap`. And looks like it'll be quite smooth - a large proportion of the API surface between the two packages are similar and require minor code juggling. And luckily we're not in too deep with `react-bootstrap` to shift it.

The most complex compoent to shift over will probably be the `PopoverField`, from a gut feel.

**Comment by chillu** [comment]
There's a separate card for the reactstrap work now: https://github.com/silverstripe/silverstripe-admin/issues/415

**Comment by newleeland** [comment]
**Potential Designs**
After trialing several issues. The most elegant solution is to just have toolbar toggle icon to switch on/off "archive" view.

Compared to other solutions, such as, a global trash can, Which would create another complex area that the team has to maintain, rather keeping the items in the context of where there were.

Opening Archive
![grid copy 14](https://user-images.githubusercontent.com/8085295/35782685-ad6d469a-0a60-11e8-91a3-79e21e9ff9d1.png)

In Archive view
![grid copy 16](https://user-images.githubusercontent.com/8085295/35782686-ad9ac674-0a60-11e8-9d46-a61917cd7020.png)

**Comment by flamerohr** [comment]
sweet, I like that idea

a bit of critique: could we have the "Archived icon is ON" point adjusted to "opened archive icon shows"?
The behaviour suggested would be inconsistent with other button behaviours, e.g. would it show any visual indicator when clicked if it's already active?

**Comment by chillu** [comment]
I've added some bigger picture comments on the (older) [Trashcan](https://github.com/silverstripe/silverstripe-framework/issues/6202) card.

**Comment by chillu** [comment]
Discussed with @clarkepaul and @newleeland, this issue is blocked until there's a UX direction for a global and separate trashcan UI, which would remove the need for specialised archive toggles in different UIs. See https://app.zenhub.com/workspace/o/silverstripe/silverstripe-framework/issues/6202

**Comment by chillu** [comment]
Same concerns as on https://github.com/silverstripe/silverstripe-versioned/issues/91

> This is partially handled by the [global archive view](https://github.com/silverstripe/silverstripe-framework/issues/6202) now. I don't think we can fully get away with without adding a toggle for "draft" and "published" records though, as designed in #38. You might have a thousand records, and only a handful of them are already published (e.g. during a large-scale review or content migration process). At the moment, you'd have to page through a dozen list views to find those records. The new global archive view won't be of help with this.

**Comment by newleeland** [comment]
This story is no longer relevant with the global archives
https://github.com/silverstripe/silverstripe-framework/issues/6202
